pkg/controller/machine/machine_controller_test.go

```diff
@@ -17,87 +17,217 @@ limitations under the License.
+		k = komega.New(k8sClient)
 
-const timeout = time.Second * 5
+		By("Setting up a new reconciler")
+		act := newTestActuator()
+		reconciler = newReconciler(mgr, act)
+
+		err = add(mgr, reconciler, "testing")
+		Expect(err).NotTo(HaveOccurred())
+
+		var mgrCtx context.Context
+		mgrCtx, mgrCtxCancel = context.WithCancel(ctx)
+		mgrStopped = make(chan struct{})
+
+		By("Starting the manager")
+		go func() {
+			defer GinkgoRecover()
+			defer close(mgrStopped)
+
+			Expect(mgr.Start(mgrCtx)).To(Succeed())
+		}()
+
+		By("Creating the namespace")
+		namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "ms-test"}}
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+	})
 
-func TestReconcile(t *testing.T) {
-	instance := &machinev1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "foo",
-			Namespace: "default",
-			Labels: map[string]string{
-				machinev1.MachineClusterIDLabel: "foo",
+	AfterEach(func() {
+		By("Deleting the machinesets")
+		Expect(cleanResources()).To(Succeed())
+
+		By("Deleting the namespace")
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+
+		By("Closing the manager")
+		mgrCtxCancel()
+		Eventually(mgrStopped, timeout).WithTimeout(20 * time.Second).Should(BeClosed())
+	})
+
+	It("Should reconcile a Machine", func() {
+		instance := &machinev1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "foo",
+				Namespace:    "default",
+				Labels: map[string]string{
+					machinev1.MachineClusterIDLabel: "foo",
+				},
 			},
-		},
-		Spec: machinev1.MachineSpec{
-			ProviderSpec: machinev1.ProviderSpec{
-				Value: &runtime.RawExtension{
-					Raw: []byte("{}"),
+			Spec: machinev1.MachineSpec{
+				ProviderSpec: machinev1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte("{}"),
+					},
 				},
 			},
-		},
-	}
+		}
 
-	// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
-	// channel when it is finished.
-	mgr, err := manager.New(cfg, manager.Options{
-		Metrics: server.Options{
-			BindAddress: "0",
-		},
-	})
-	if err != nil {
-		t.Fatalf("error creating new manager: %v", err)
-	}
-	c = mgr.GetClient()
+		By("Creating the Machine")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
 
-	a := newTestActuator()
-	recFn := newReconciler(mgr, a)
-	if err := add(mgr, recFn, "dummy"); err != nil {
-		t.Fatalf("error adding controller to manager: %v", err)
-	}
+		By("Setting the AuthoritativeAPI to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMachineAPI
+		})).Should(Succeed())
+
+		Eventually(func() (machinev1.MachineStatus, error) {
+			machine := &machinev1.Machine{}
+			err := k8sClient.Get(ctx, objectKey(instance), machine)
+			if err != nil {
+				return machinev1.MachineStatus{}, err
+			}
+			return machine.Status, nil
+		}, timeout).ShouldNot(Equal(machinev1.MachineStatus{}))
```

Comment:
I think you could just do
```suggestion
		Eventually(k.Get(instance), timeout).Should(HaveField("Status", Not(Equal(machinev1.MachineStatus{}))))
```